creme/linear_model/ffm.py

```diff
@@ -0,0 +1,364 @@
+import collections
+import functools
+import itertools
+import numbers
+import numpy as np
+
+from .. import base
+from .. import optim
+from .. import utils
+
+
+__all__ = [
+    'FFMClassifier',
+    'FFMRegressor'
+]
+
+
+class FFM:
+    """Field-aware Factorization Machines.
+
+    Parameters:
+        n_factors (int): Dimensionality of the factorization or number of latent factors.
+        loss (optim.Loss): The loss function to optimize for.
+        optimizer (optim.Optimizer): The sequential optimizer used for updating the weights and
+            latent factors. Note that the intercept is handled separately.
+        intercept (float): Initial intercept value.
+        intercept_lr (optim.schedulers.Scheduler or float): Learning rate scheduler used for
+            updating the intercept. If a `float` is passed, then an instance of
+            `optim.schedulers.Constant` will be used. Setting this to 0 implies that the intercept
+            will be not be updated.
+        weight_initializer (optim.initializers.Initializer): Weights initialization scheme. Defaults
+            to ``optim.initializers.Zeros()``.
+        latent_initializer (optim.initializers.Initializer): Latent factors initialization scheme.
+            Defaults to ``optim.initializers.Normal(mu=.0, sigma=.1, random_state=random_state)``.
+        l1 (float): Amount of L1 regularization used to push weights towards 0.
+        l2 (float): Amount of L2 regularization used to push weights towards 0.
+        clip_gradient (float): Clips the absolute value of each gradient value.
+        random_state (int, ``numpy.random.RandomState`` instance or None): If int, ``random_state``
+            is the seed used by the random number generator; if ``RandomState`` instance,
+            ``random_state`` is the random number generator; if ``None``, the random number
+            generator is the ``RandomState`` instance used by `numpy.random`.
+
+    Attributes:
+        weights (collections.defaultdict): The current weights assigned to the features.
+        latents (collections.defaultdict): The current latent weights assigned to the features.
+
+    """
+
+    def __init__(self, n_factors, loss, optimizer, intercept, intercept_lr, weight_initializer,
+                 latent_initializer, l1, l2, clip_gradient, random_state):
+        self.n_factors = n_factors
+        self.loss = loss
+        self.optimizer = optim.SGD(0.01) if optimizer is None else optimizer
+        self.intercept = intercept
+
+        self.intercept_lr = (
+            optim.schedulers.Constant(intercept_lr)
+            if isinstance(intercept_lr, numbers.Number) else
+            intercept_lr
+        )
+
+        if weight_initializer:
```

Comment:
A cleaner way here would be:

```python
if weight_initializer is None:
    weight_initializer = optim.initializers.Zeros()
self.weight_initializer = weight_initializer
```

But I'm probably nitpicking :)

Reply:
Nice catch!